### ISSUE TITLE
[Feature Request]: Dependancy Managment Standarization

### ISSUE THREAD

**Issue by KGB33** [SpokaneTech/SpokaneTech_Py#121]
### Description

Currently, `SpokaneTech_Py` is using a vanilla `requirements.txt` to manage dependencies. I think the project has grown enough that it's time to migrate to a (slightly) more robust system.

While PR #1 proposed a Poetry based system, I've found that a `pip-compile` biased workflow works better for applications. Plus, it only requires `uv`/`pip-tools` to be installed when updating/adding dependencies; `pip` will still work for installing the dependencies.

To migrate, `requirements.txt` and `requirements/dev.txt` will be removed, and moved into `pyproject.toml` under the (new) `[project.dependencies]` and `[project.optional-dependencies]` tables respectively. Then, generate a lock-file using `uv pip compile pyproject.toml -o requirements.lock` (and optionally, lock the development dependencies too: `uv pip compile pyproject.toml --extra dev -o requirements.dev.lock`)

This will slightly change the developer experience, instead of `pip install -r requirements.txt -r requirements/dev.txt` it'll be `pip install -e .[dev]` (or `pip install -r uv.dev.lock`). Plus, Docker builds will be more reproducible.

Additionally, two more tables will need to be added to `pyproject.toml`: `[build-system]` (I prefer Hatchling), and `[project]` (which we should probably have anyways).

TLDR:
Move to a `uv`/`pip-tools` workflow by:
  - [ ] Move `requirements.txt` and `requirements/dev.txt` into `pyproject.toml`
  - [ ] Generate `requirements.lock` and `requirements.dev.lock` (names are flexible)
  - [ ] Add `[project]` and `[build-system]` tables to `pyproject.toml`
  - [ ] Update getting started docs

---

I migrated one of my personal projects to this system about a month ago and have been happy with it so far ([reference PR here](https://github.com/KGB33/RoboShpee/pull/25)). 

I'd appreciate any suggestions or alternatives.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by joeriddles** [comment]
@KGB33 this is great. I've used `pip-tools` before and liked it. I have no used `uv`, does it wrap `pip-tools` or just implement similar commands?

Let's do this! Thanks for writing up this issue.

**Comment by KGB33** [comment]
>  I have not used uv, does it wrap pip-tools or just implement similar commands?

It just implements similar commands (in rust). Its from the same folks as `ruff`. At a quick glance it seems like the lockfiles are almost exactly the same.
